### ISSUE TITLE
dropwizard-example maven-enforcer-plugin Dependency Convergence outputs invalid results

### ISSUE THREAD

**Issue by BeniaminK** [dropwizard/dropwizard#4196]
I was checking the `dropwizard-example` project and I saw that it uses very old `1.4.1` version of `maven-enforcer-plugin`. When I updated the version to `3.0.0` (the latest one) - the DependencyConvergency task started to fail and indeed the dependency convergency fails. I think this should be updated and either `<DependencyConvergence />` rule should be removed or `dropwizard-dependencies` should point to the correct artifacts.

**Comment by BeniaminK** [comment]
```
[INFO] --- maven-enforcer-plugin:3.0.0:enforce (enforce) @ dropwizardtest ---
[WARNING]
Dependency convergence error for org.osgi:org.osgi.core:jar:6.0.0:provided paths to dependency are:
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.glassfish.jersey.core:jersey-common:jar:2.33:compile
      +-org.osgi:org.osgi.core:jar:6.0.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.hibernate.validator:hibernate-validator:jar:6.1.7.Final:compile
      +-org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile
        +-org.apache.logging.log4j:log4j-api:jar:2.0:provided
          +-org.osgi:org.osgi.core:jar:4.3.1:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-inmemory:jar:2.33:test
    +-org.glassfish.jersey.core:jersey-server:jar:2.33:compile
      +-org.osgi:org.osgi.core:jar:6.0.0:provided

[WARNING]
Dependency convergence error for org.yaml:snakeyaml:jar:1.26:compile paths to dependency are:
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard:dropwizard-configuration:jar:2.0.24:compile
      +-com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.5:compile
        +-org.yaml:snakeyaml:jar:1.26:compile
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-inmemory:jar:2.33:test
    +-org.glassfish.jersey.test-framework:jersey-test-framework-core:jar:2.33:test
      +-org.testng:testng:jar:6.9.6:test
        +-org.yaml:snakeyaml:jar:1.15:test

[WARNING]
Dependency convergence error for javax.servlet:javax.servlet-api:jar:3.0.1:provided paths to dependency are:
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.prometheus:simpleclient_servlet:jar:0.11.0:compile
    +-javax.servlet:javax.servlet-api:jar:3.0.1:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard:dropwizard-jetty:jar:2.0.24:compile
      +-org.eclipse.jetty:jetty-servlets:jar:9.4.43.v20210629:compile
        +-org.eclipse.jetty:jetty-continuation:jar:9.4.43.v20210629:compile
          +-javax.servlet:javax.servlet-api:jar:3.1.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard:dropwizard-jetty:jar:2.0.24:compile
      +-org.eclipse.jetty:jetty-servlets:jar:9.4.43.v20210629:compile
        +-javax.servlet:javax.servlet-api:jar:3.1.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard:dropwizard-jetty:jar:2.0.24:compile
      +-org.eclipse.jetty:jetty-http:jar:9.4.43.v20210629:compile
        +-javax.servlet:javax.servlet-api:jar:3.1.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard.metrics:metrics-servlets:jar:4.1.25:compile
      +-javax.servlet:javax.servlet-api:jar:3.1.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.eclipse.jetty:jetty-servlet:jar:9.4.43.v20210629:compile
      +-org.eclipse.jetty:jetty-util-ajax:jar:9.4.43.v20210629:compile
        +-javax.servlet:javax.servlet-api:jar:3.1.0:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.eclipse.jetty:jetty-util:jar:9.4.43.v20210629:compile
      +-javax.servlet:javax.servlet-api:jar:3.1.0:provided

[WARNING]
Dependency convergence error for org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile paths to dependency are:
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.hibernate.validator:hibernate-validator:jar:6.1.7.Final:compile
      +-org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.hibernate.validator:hibernate-validator:jar:6.1.7.Final:compile
      +-org.jboss.logging:jboss-logging-processor:jar:2.1.0.Final:provided
        +-org.jboss.logging:jboss-logging:jar:3.1.2.GA:provided
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-org.hibernate.validator:hibernate-validator:jar:6.1.7.Final:compile
      +-org.jboss.logging:jboss-logging-annotations:jar:2.1.0.Final:provided
        +-org.jboss.logging:jboss-logging:jar:3.1.2.GA:provided

[WARNING] Rule 1: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.053 s
[INFO] Finished at: 2021-08-18T17:42:32+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by joschi** [comment]
@BeniaminK Thanks for reporting this!

Would you be able to provide a pull request addressing the issue?

See also:
- https://github.com/dropwizard/dropwizard/pull/4181
- https://github.com/dropwizard/dropwizard/pull/4182
- https://github.com/dropwizard/dropwizard/pull/4183
- https://github.com/dropwizard/dropwizard/pull/4184

**Comment by joschi** [comment]
I'm also contemplating just skipping the upgrade to maven-enforcer-plugin 3.0.0 because it mixes up compile/runtime and test dependencies:
```
[WARNING]
Dependency convergence error for org.yaml:snakeyaml:jar:1.26:compile paths to dependency are:
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-io.dropwizard:dropwizard-core:jar:2.0.24:compile
    +-io.dropwizard:dropwizard-configuration:jar:2.0.24:compile
      +-com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.10.5:compile
        +-org.yaml:snakeyaml:jar:1.26:compile
and
+-com.bendzi.dropwizmetrics:dropwizardtest:jar:0.0.1
  +-org.glassfish.jersey.test-framework.providers:jersey-test-framework-provider-inmemory:jar:2.33:test
    +-org.glassfish.jersey.test-framework:jersey-test-framework-core:jar:2.33:test
      +-org.testng:testng:jar:6.9.6:test
        +-org.yaml:snakeyaml:jar:1.15:test
```

**Comment by josephlbarnett** [comment]
tracked for enforcer plugin here: https://issues.apache.org/jira/browse/MENFORCER-394

**Comment by natalie-zamani** [comment]
It looks like a fix was recently committed fixing this, but I don't see it released yet: 
https://gitbox.apache.org/repos/asf?p=maven-enforcer.git;a=commit;h=b20f9114d67bd9eb7957b3e2c77ba99fcfe8efe1

**Comment by joschi** [comment]
Closed via #5413.
